src/System.Memory/tests/Memory/Equality.cs

```diff
@@ -120,6 +120,58 @@ public static void DefaultMemoryCanBeBoxed()
             Assert.True(readOnlyMemory.Equals(memoryAsObject));
         }
 
+        [Fact]
+        public static void EqualityThroughInterface_True()
+        {
+            int[] array = { 42, 43, 44, 45, 46 };
+            IEquatable<Memory<int>> left = new Memory<int>(array, 2, 3);
+            IEquatable<Memory<int>> right = new Memory<int>(array, 2, 3);
+
+            Assert.True(left.Equals(right));
```

Comment:
Thinking about it again, this resolves to `object.Equals(object)` call. I think the test for type mismatch should go here instead, actually.